src/components/Profile/ProfileForm.js

```diff
@@ -0,0 +1,418 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import {
+  Box,
+  Button,
+  ButtonBase,
+  Field,
+  GU,
+  IconCheck,
+  TextInput,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@1hive/1hive-ui'
+import SingleDatePicker from '../SingleDatePicker/SingleDatePicker'
+import { useProfile } from '../../providers/Profile'
+
+import { dateFormat } from '../../utils/date-utils'
+import { validateEmail } from '../../utils/validate-utils'
+
+function ProfileForm({ fetchProfilePic, onBack, profilePic }) {
+  const { name: layout } = useLayout()
+  const {
+    birthday,
+    description,
+    email,
+    location,
+    name,
+    updateProfile,
+    verifiedAccounts,
+    website,
+  } = useProfile()
+  const [formData, setFormData] = useState({
+    birthday,
```

Comment:
In the case of public profile data, we can assume it's already present because we are not letting the users edit their profile if they haven't opened their profile box, and for that to happen public data should have already loaded.

For private data i think it makes sense to do handle this case yeah, but i would suggest using a `useEffect` where we update the `formData` state when comes new private data. What do you think?

Reply:
Done